src/main/java/WoopAI/Parser.java

```diff
@@ -0,0 +1,83 @@
+package WoopAI;
+
+/**
+ * Class to handle all user input and return needed information.
+ */
+public class Parser {
+    /**
+     * Retrieves the command from the given input.
+     *
+     * @param input The user input.
+     * @return The command part of the input.
+     */
+    public static String parseCommand(String input) {
+        return input.split(" ")[0];
+    }
+
+    /**
+     * Retrieves the Task descriptor from the user input depending on the task type.
+     * Additionally, creates a Task object that represents the descriptor.
+     * If descriptor is invalid, throws IllegalDescriptorException.
+     *
+     * @param input The user input.
+     * @param type The Task type that this descriptor is for.
+     * @return A new Task object that represents the descriptor.
+     * @throws IllegalDescriptorException If descriptor syntax is invalid.
+     * @throws UnknownCommandException If type given is invalid.
+     */
+    public static Task parseDescriptor(String input, TaskType type)
+            throws IllegalDescriptorException, UnknownCommandException {
+        int index = input.indexOf(" ");
+        if (index >= 0 && (index + 1) < input.length()) {
```

Comment:
```suggestion
        if (0 <= index && index < input.length() - 1) {
```
This is subjective, but I personally find this to be more readable and clearer, feel free to discuss.